inc/functions.php

```diff
@@ -569,49 +627,44 @@ function my_mail($to, $subject, $message, $from="", $charset="", $headers="", $k
 	global $mybb, $plugins;
 	static $mail;
 
-	// Does our object not exist? Create it
-	if(!is_object($mail))
-	{
-		require_once MYBB_ROOT."inc/class_mailhandler.php";
+	// Get our mail handler.
+	$mail = &get_mybb_mailhandler();
 
-		$plugins->run_hooks('my_mail_mailhandler_init', $mail);
-
-		if(!is_object($mail) || !($mail instanceof MailHandler))
+	$my_mail_parameters = compact("to", "subject", "message", "from", "charset", "headers", "keep_alive", "format", "message_text", "return_email");
```

Comment:
Is it necessary to do here, letting the plugin to override values?
Let the plugin do what it wants, we are anyway filling the rest in line 646

Reply:
> Is it necessary to do here, letting the plugin to override values?
> Let the plugin do what it wants, we are anyway filling the rest in line 646

It's for a dedicated hook, for only processing mail parameters.